src/app/tabs/tabbed-menu.js

```diff
@@ -1,52 +1,97 @@
 var yo = require('yo-yo')
-var helper = require('../../lib/helper')
+var csjs = require('csjs-inject')
+var remixLib = require('remix-lib')
 
-var css = require('./styles/tabbed-menu-styles')
+var styles = require('../ui/styles-guide/theme-chooser').chooser()
 
-class TabbedMenu {
-  constructor (api = {}, events = {}, opts = {}) {
-    var self = this
-    var tabView = document.createElement('ul')
-    this.tabView = tabView
-    this.events = events
-    this.tabs = {}
-    this.contents = {}
-
-    events.app.register('debuggingRequested', () => {
-      self.selectTab(tabView.querySelector('li.debugView'))
-    })
+var EventManager = remixLib.EventManager
 
-    events.rhp.register('switchTab', tabName => {
```

Comment:
events registration has to be kept. that might be the reason why tests are failing

Reply:
oh i see.

I think it's not completely clear to me when we use events and when we use methdod calls.

My understanding is, that events are fired, when we don't know who is interested in them.
Then components who want to know, will register for a certain event.

When a "switchTab" is triggered, i don't see it as a notification of RHP to the rest of the world,
but I see it as a request - not to the world, but actually only to the tabbed-menu specifically.
The RHP even created and has access to the tabbed-menu directly, so I feel a method should be called instead of doing it through events.

I think I would prefer the RHP to just call `tabbedMenu.switchTab(tabName)`

Reply:
ok I see . yes it make sense
so in that case you should update this part https://github.com/ethereum/remix-ide/blob/1c0472f0e806b39fdebbcf7aa619b670ec159ff8/src/app/panels/righthand-panel.js#L41